src/util/subscribeToResult.ts

```diff
@@ -51,9 +63,9 @@ export function subscribeToResult<T, R>(outerSubscriber: OuterSubscriber<T, R>,
       root.setTimeout(() => { throw err; });
     });
     return destination;
-  } else if (typeof result[$$iterator] === 'function') {
-    for (let item of result) {
-      destination.next(item);
+  } else if (isIterable(result)) {
```

Comment:
There was probably no reason to break this out into it's own function. It's only used in one spot. Also, you're forced to close over the other function now, *and* it's an additional function allocation.